### PR TITLE
Use `pyproject.toml` in the plugin examples

### PR DESCRIPTION
**Pull request by LecrisUT**
Not sure how much I can design this, ultimately I want it to address #2810, but for now there are some low-hanging fruits:
- Use `pyproject.toml` in the examples

### CODE REVIEW

**Review by lukaszachy — APPROVED** [review]

**Review by psss — APPROVED** [review]
Thanks for improving this!

**Comment by happz** [comment]
@LecrisUT there seems to be some issue on CentOS Stream 9, see https://artifacts.dev.testing-farm.io/4c90b5bc-8174-4c64-b7e3-8c5f392d5bcc/work-advancedjytot00y/plans/features/advanced/execute/data/guest/default-0/tests/plugins-15/output.txt. Maybe older-than-expected packaging infrastructure, not supporting the feature?

**Comment by LecrisUT** [comment]
Weird, why does it give `UNKNOWN`, just double-checked and I did populate `name`. Maybe it's the lack of `build-system.build-backend`. Can you try out quickly with `setuptools` or `hatchling`, otherwise I'll try it next week (still on train rn)

**Comment by happz** [comment]
> Weird, why does it give `UNKNOWN`, just double-checked and I did populate `name`. Maybe it's the lack of `build-system.build-backend`. Can you try out quickly with `setuptools` or `hatchling`, otherwise I'll try it next week (still on train rn)

No worries, next week is absolutely fine. It's Saturday after all. I just wanted to put a status update so you're aware.

**Comment by LecrisUT** [comment]
Note to self: add `--no-index` to the `pip install`. Just as a sanity check that all dependencies are from the system. What about `--no-build-isollation` as well?

**Comment by LecrisUT** [comment]
Ok, seems `setuptools 53.0.0` is just unusable for PEP517, `hatchling` still works though, so might as well switch the backend.

File: examples/plugins/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[build-system]
+# Using hatchling as build backend because EPEL9 setuptools (53.0.0) does not build with PEP517 properly
```
[the remark this answers is not in the document]
Comment (happz) [reply examples/plugins/pyproject.toml:2]: Sounds good to me.

**Review by thrix — APPROVED** [review]
LGTM

**Review by martinhoyer — APPROVED** [review]

**Pull request merged**